osconfig_tests/test_suites/patch/patch.go

```diff
@@ -133,39 +210,74 @@ func runExecutePatchJobTest(ctx context.Context, testCase *junitxml.TestCase, te
 	}
 
 	testCase.Logf("Started patch job '%s'", job.GetName())
+	if err := waitPatchJob(ctx, job, testSetup.assertTimeout); err != nil {
+		testCase.WriteFailure("Patch job '%s' error: %v", job.GetName(), err)
+	}
+}
 
-	logger.Printf("%v\n", job)
+func runRebootPatchTest(ctx context.Context, testCase *junitxml.TestCase, testSetup *patchTestSetup, testProjectConfig *testconfig.Project, pc *osconfigpb.PatchConfig, minBootCount, maxBootCount int) {
+	client, err := daisyCompute.NewClient(ctx)
+	if err != nil {
+		testCase.WriteFailure("error creating client: %v", err)
+		return
+	}
 
-	// assertion
-	tick := time.Tick(10 * time.Second)
-	timedout := time.Tick(testSetup.assertTimeout)
-	for {
-		select {
-		case <-timedout:
-			testCase.WriteFailure("Patch job '%s' timed out", job.GetName())
-			return
-		case <-tick:
-			req := &osconfigpb.GetPatchJobRequest{
-				Name: job.GetName(),
-			}
-			res, err := osconfigClient.GetPatchJob(ctx, req)
-			if err != nil {
-				testCase.WriteFailure("Error while fetching patch job: \n%s\n", utils.GetStatusFromError(err))
-				return
-			}
+	testCase.Logf("Creating instance with image %q", testSetup.image)
+	var metadataItems []*api.MetadataItems
+	metadataItems = append(metadataItems, testSetup.startup)
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("os-config-enabled-prerelease-features", "ospatch"))
+	name := fmt.Sprintf("patch-reboot-%s-%s", path.Base(testSetup.testName), suffix)
+	inst, err := utils.CreateComputeInstance(metadataItems, client, testSetup.machineType, testSetup.image, name, testProjectConfig.TestProjectID, testProjectConfig.GetZone(), testProjectConfig.ServiceAccountEmail, testProjectConfig.ServiceAccountScopes)
+	if err != nil {
+		testCase.WriteFailure("Error creating instance: %v", utils.GetStatusFromError(err))
+		return
+	}
+	defer inst.Cleanup()
 
-			if isPatchJobFailureState(res.State) {
-				testCase.WriteFailure("Patch job '%s' completed with status %v and message '%s'", job.GetName(), res.State, job.GetErrorMessage())
-				return
-			}
+	testCase.Logf("Waiting for agent install to complete")
+	if _, err := inst.WaitForGuestAttribute("osconfig_tests/", "install_done", 5*time.Second, 5*time.Minute); err != nil {
+		testCase.WriteFailure("Error waiting for osconfig agent install: %v", err)
+		return
+	}
 
-			if res.State == osconfigpb.PatchJob_SUCCEEDED {
-				if res.InstanceDetailsSummary.GetInstancesSucceeded() < 1 {
-					testCase.WriteFailure("Patch job '%s' completed with no instances patched", job.GetName())
-				}
-				return
-			}
-		}
+	testCase.Logf("Agent installed successfully")
+
+	parent := fmt.Sprintf("projects/%s", testProjectConfig.TestProjectID)
+	osconfigClient, err := gcpclients.GetOsConfigClient(ctx)
+
+	req := &osconfigpb.ExecutePatchJobRequest{
+		Parent:      parent,
+		Description: "testing reboot patch job run",
```

Comment:
nit: can this be reworded to "testing patch job reboot"? That should make it clearer in the logs.